### ISSUE TITLE
Custom logging level to Application Insight does not work as expected

### ISSUE THREAD

**Issue by yli02** [Azure/azure-sdk-for-python#33980]
- **Package Name**: azure-monitor-opentelemetry
- **Package Version**:  1.2.0
- **Operating System**: python:3.8.18-slim
- **Python Version**:  python 3.8.18

**Describe the bug**
The documentation at [1] in footnote 4 told that we could use the reference link to setup logging level to send to Application Insight. I have tried to set `APPLICATIONINSIGHTS_INSTRUMENTATION_LOGGING_LEVEL="INFO"` to environment variable, however, it still sends debug log to Application Insights, as the instrument logger is set with logging level `DEBUG`. I also tried to set `OTEL_LOG_LEVEL` to `INFO`, which also don't work, debug logs are still sent to Application Insights.

**To Reproduce**
1. Create a logger from the standard python logging library with a name
2. Set the logging level from 1st step created logger to `DEBUG`
3. Use `configure_azure_monitor` function to bind to the logger created in 1st step
4. Set the environment variables like described in the ***Describe the bug** section
5. Logs send with `debug` method from 1st step created logger to be sent to Application Insights.

**Expected behavior**
Log level specified to send to Application Insights should be effective.

**Screenshots**

**Additional context**
The `python:3.8.18-slim` is a docker image.

[1] https://learn.microsoft.com/en-us/azure/azure-monitor/app/opentelemetry-add-modify?tabs=python%2Cpython-1%2Cpython-2%2Cpython-3%2Cpython-4%2Cpython-5%2Cpython-6%2Cpython-8%2Cpython-9%2Cpython-10%2Cpython-11%2Cpython-12%2Cpython-13%2Cpython-14%2Cpython-15#included-instrumentation-libraries
[2] https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/#general-sdk-configuration

**Comment by kristapratico** [comment]
@yli02 thanks for your issue, the library owners will take a look soon and respond.

**Comment by jeremydvoss** [comment]
OTEL_LOG_LEVEL is not implemented yet by opentelemetry-python due to a [spec dispute](https://github.com/open-telemetry/opentelemetry-specification/issues/920)
APPLICATIONINSIGHTS_INSTRUMENTATION_LOGGING_LEVEL is an environment variable for the Java SDK not Python. (I will talk to our docs team to make this clearer)
Please see our [README ](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/monitor/azure-monitor-opentelemetry/README.md)and [configuration docs](https://learn.microsoft.com/en-us/azure/azure-monitor/app/opentelemetry-configuration?tabs=python) for more configuration options.

I think I understand your confusion, though. In opentelemetry-python, the OTel logging is built on native python logging which is [hierarchical](https://docs.python.org/3/library/logging.html#logger-objects). Take the following example:
```
configure_azure_monitor()

logger1 = getLogger("foo1.bar")
logger1.setLevel(INFO)
logger1.debug("foo1.bar debug")
logger1.info("foo1.bar info")

logger2 = getLogger("foo2.bar")
logger2.setLevel(WARN)
logger2.debug("foo2.bar debug")
logger2.info("foo2.bar info")
logger2.warn("foo2.bar warn")
```
Because no logger_name was passed in, the root logger is instrumented. This means all logs are captured _if_ they pass their respective logger's levels. So, `foo1.bar info` and `foo2.bar warn` are captured. If you wanting to limit captured logs to a specific module, you can specify logger_name:
```
configure_azure_monitor(logger_name="foo1")

logger1 = getLogger("foo1.bar")
logger1.setLevel(INFO)
logger1.debug("foo1.bar debug")
logger1.info("foo1.bar info")

logger2 = getLogger("foo2.bar")
logger2.setLevel(WARN)
logger2.debug("foo2.bar debug")
logger2.info("foo2.bar info")
logger2.warn("foo2.bar warn")
```
In this case, only `foo1.bar info` is captured. Let me know if this clears things up.

**Comment by yli02** [comment]
@jeremydvoss thanks a lot for your explanations.

We actually configured the monitoring to be the 2nd case you mentioned. But I was thinking that during development time, I could set `logger2.setLevel(DEBUG)`, but configure an env variable like `OTEL_LOG_LEVEL` to be `INFO`, so that I could still check the debugging logs from the stdout, but only get logs equal to or above `INFO` to be send to Application Insights. To me, this sounds what should have been supported based on the footnote 4 at [1], right? or I misinterpreted it? Thanks.

[1] https://learn.microsoft.com/en-us/azure/azure-monitor/app/opentelemetry-add-modify?tabs=python%2Cpython-1%2Cpython-2%2Cpython-3%2Cpython-4%2Cpython-5%2Cpython-6%2Cpython-8%2Cpython-9%2Cpython-10%2Cpython-11%2Cpython-12%2Cpython-13%2Cpython-14%2Cpython-15#included-instrumentation-libraries
